students/ScottL/session09/cmi_main.py

```diff
@@ -0,0 +1,155 @@
+            print(self.DONOR_LIST_HEADER)
+            for name in md.DonorCollection().name_list():
+                print(name)
+        else:
+            print("\n\tThe name list is empty.")
+
+    def validate_name(self):
+        """
+        Process the expected user input
+        :return: 2-element tuple containing the first and last names
+        """
+        first = self.get_donor_name("FIRST")
+        if first.strip().lower() not in ("quit", ""):
+            if first.strip().lower() == "list":
+                self.display_donor_list()
+            else:
+                last = self.get_donor_name("LAST")
+                if last.strip().lower() == "list":
+                    self.display_donor_list()
+                else:
+                    return first, last
+
+    @staticmethod
+    def get_donation_amount():
+        """
+        Receive and process input from user on the donation amount.
+        :return: float-type number of donation amount
+        """
+        prompt = "Enter the amount of the donation: "
+        amount = input(prompt)
+        value_error_message = "\nThe entry for 'amount' count was not a valid number. " \
+                              "Please try again."
+        if amount.strip().lower() not in ("quit", ""):
+            try:
+                amount = "".join(amount.split(","))
+                amount = float(amount.strip("$"))
+            except ValueError:
+                print(value_error_message)
+            else:
+                if amount < 0:
+                    print("\nUser attempted to enter a negative number; "
+                          "please input a positive number for the donation.")
+                else:
+                    return amount
+
+    def new_donation(self):
+        """Organize process flow for calling and validating user input for new donations"""
+        try:
+            first, last = self.validate_name()
+        except TypeError:
+            self.main_menu()
+        else:
+            if last.lower() not in ("quit", ""):
+                first = first.capitalize()
+                last = last.capitalize()
+                amount = self.get_donation_amount()
+                if amount is not None:
+                    print(md.DonorCollection().new_donation(first, last, amount))
+
+    def report(self):
+        """Display donation data report to user."""
+        print(self.REPORT_HEADER_1, self.REPORT_HEADER_2)
+        report = md.DonorCollection().report()
+        if report:
+            for item in report:
+                print("{:<26}${:>12,}{:^17}${:>11,}".format(item[0], item[1], item[2], item[3]))
+        else:
+            print("\t\t\t*** No data exists to display ***")
+
+    @staticmethod
+    def email():
+        """Write text(s) to file simulating email(s) thanking donor(s)."""
+        emails = md.DonorCollection().group_email()
+        for email in emails:
+            try:
+                with open(f"{email}.txt", "w") as file:
```

Comment:
`file` is a python keyword.  You should change it to `my_file` or something else.